### PR TITLE
feat: tag condition type for README contents

### PR DESCRIPTION
**Pull request by Chriscbr**
Introduces a new tag condition type that applies tags based on whether a string appears in the README of packages.

Example:
```ts
TagCondition.readme().includes('ECS');
TagCondition.readme().includes('fargate', { atLeast: 3, caseSensitive: false });
```

Part of #625

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by gabewomble — APPROVED** [review]
Looks nice! Specifying options for atLeast and caseSensitive with sane defaults is a great inclusion as well. I've added do-not-merge to give other reviewers time to take a look if needed

**Review by MrArnoldPalmer — COMMENTED** [review]
what do you think of changing the signature of `isTagApplicable` to:
```ts
isTagApplicable(config: TagConditionConfig, { pkg: object, readme: string }): boolean
```
Additionally changing all of the other utility functions underneath so as we add more values that can be targeted with this comparison logic we aren't continually adding arguments.

Also to keep finding the value to compare, and the comparison logic separate, we should change `getNestedField` to accept `TagConditionConfig` as it's first argument so it can just return whatever value is used on the left side of the comparison. IE, the field value from package,json or the readme string. Though get nested field on it's own is still useful so another function above that like `getPackageValue` or something.

Basically I think `TagConditionLogicType.INCLUDES` should be able to operate on the readme string or any other string the user wants to compare with. this would require changing the `comparison` type signature to pass a third argument of `options` or `config` relevant to the comparison logic. Just thinking about continuing to add conditions and values to be targeted without creating new one off operators every time. WDYT?

```ts
type comparison = (field: any, value: any, options) => boolean;
const fnMap: LogicMap = {
    [TagConditionLogicType.INCLUDES]: createFieldComparator((field, val, options) => // do all the comparison logic, or pass a function that does,
    // ...
  };

**Review by MrArnoldPalmer — APPROVED** [review]
